### PR TITLE
Problem: There is no support for request nor timeout

### PR DESCRIPTION
**Pull request by aquette**
Solution: Implement these

### CODE REVIEW

File: src/bmsg.c [hunk 1]
```diff
@@ -585,6 +601,73 @@ int main (int argc, char *argv [])
...
+        char *agent_name = argv[argn];
+        if (!agent_name)
+            die ("missing agent name", NULL);
+
+        argn ++;
+        char *subject = argv[argn];
+        if (!subject)
+            die ("missing subject", NULL);
+
+        mlm_client_set_producer (client, subject);
```
Comment (geraldguillaume) [inline src/bmsg.c:616]: as this is relative to STREAM, it is not appropriate here

File: src/bmsg.c [hunk 2]
```diff
@@ -585,6 +601,73 @@ int main (int argc, char *argv [])
...
+
+        zmsg_t *msg = zmsg_new ();
+        zmsg_addstr (msg, "GET");
```
Comment (geraldguillaume) [inline src/bmsg.c:619]: Adding "GET" in the payload is a mistake, because it is dependant to the protocol itself

File: src/bmsg.c [hunk 3]
```diff
@@ -585,6 +601,73 @@ int main (int argc, char *argv [])
...
+
+            const char *sender = mlm_client_sender (client);
+            const char *subj = mlm_client_subject (client);
+            if (streq (sender, agent_name)) {
+                if (streq (subj, subject)) {
+                    if ( verbose ) {
+                        zsys_debug ("Got reply");
+                        zmsg_print (msg);
+                    }
+                    char *okfail = zmsg_popstr (msg);
```
Comment (geraldguillaume) [inline src/bmsg.c:654]: My recommendation is to do not consider the payload result. Just print it !

File: src/bmsg.c [hunk 4]
```diff
@@ -585,6 +601,73 @@ int main (int argc, char *argv [])
...
+                    if ( okfail && streq (okfail, "OK")) {
+                        char *rc = zmsg_popstr(msg);
+                        while (rc) {
```
Comment (geraldguillaume) [inline src/bmsg.c:657]: As you already print the replied message (zmsg_print l652), I don't see the interest to print it again

**Comment by aquette** [comment]
FYI, I'm preparing an updated PR with the improvements pointed by @geraldguillaume 
As a general note, beside from allowing to request for test & dev purposes, the idea is also to have an accessor that allows for local scripting.

**Pull request merged**